### PR TITLE
Add ZSA Moonlander

### PR DESCRIPTION
**Pull request by MiddleMan5**
## Description

Add ZSA Moonlander MK1 support

## QMK Pull Request 
https://github.com/qmk/qmk_firmware/pull/14603

## Checklist

- [ ] The VIA support for this keyboard is in QMK master already **(MANDATORY)**
- [x] The VIA definition follows the guide here: https://caniusevia.com/docs/layouts
- [x] I have tested this keyboard definition using VIA's "Design" tab.
- [x] I have tested this keyboard definition with firmware on a device.
- [x] I have assigned alpha keys and modifier keys with the correct colors.
- [x] The Vendor ID is not `0xFEED`

Tested with VIA:
image

### CODE REVIEW

File: src/zsa/moonlander.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
+{
+  "name": "Moonlander Mark I",
+  "vendorId": "0x3297",
+  "productId": "0x1969",
+  "lighting": "qmk_rgblight",
```
Comment (drashna) [inline src/zsa/moonlander.json:5]:
```suggestion
    "lighting": {
      "extends": "qmk_rgblight",
      "keycodes": "qmk",
      "underglowEffects": [
        ["Off", 0],
        ["Solid Color", 1],
        ["Alpha Mods", 1],
        ["Gradient Up-Down", 1],
        ["Gradient Left-Right", 1],
        ["Breathing", 1],
        ["Band Saturation", 1],
        ["Band Brightness", 1],
        ["Band Pinwheel Saturation", 1],
        ["Band Pinwheel Brightness", 1],
        ["Band Spiral Saturation", 1],
        ["Band Spiral Brightness", 1],
        ["Gradient", 1],
        ["Gradient Cycle Left-Right", 1],
        ["Gradient Cycle Up-Down", 1],
        ["Cycle In", 1],
        ["Cycle In (Dual)", 1],
        ["Rainbow Chevron", 1],
        ["Gradient Pinwheel", 1],
        ["Gradient Spiral", 1],
        ["Gradient Dual Beacon", 1],
        ["Gradient Beacon", 1],
        ["Dual Pinwheel", 1],
        ["Raindrops", 1],
        ["Jellybean Raindrops", 1],
        ["Hue Breathing", 1],
        ["Hue Bendulum", 1],
        ["Hue Wave", 1],
        ["Heatmap", 1],
        ["Digitial Rain", 1],
        ["Reactive (Simple)", 1],
        ["Reactive", 1],
        ["Reactive Wide", 1],
        ["Reactive Wide (Multi)", 1],
        ["Reactive Cross", 1],
        ["Reactive Cross (Multi)", 1],
        ["Reactive Nexus", 1],
        ["Reactive Nexus (Multi)", 1],
        ["Splash (Multi-Color)", 1],
        ["Splash (Multi-Color) (Multi)", 1],
        ["Splash (Solid Color)", 1],
        ["Splash (Solid Color) (Multi)", 1]
      ]
    },
```

File: src/zsa/moonlander.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        "5,2",
+        {"x": 0.5, "h": 2},
+        "11,4",
+        {"h": 2},
+        "11,5",
+        {"h": 2},
+        "11,6"
+      ],
+      [{"y": -0.875, "x": 0.5}, "4,0", "4,1", {"x": 14.5}, "10,5", "10,6"]
+    ]
```
Comment (drashna) [inline src/zsa/moonlander.json:96]:
```suggestion
      [
        {
          "x": 3.5
        },
        "0,3",
        {
          "x": 11.5
        },
        "6,3"
      ],
      [
        {
          "y": -0.875,
          "x": 2.5
        },
        "0,2",
        {
          "x": 1
        },
        "0,4"
      ],
      [
        {
          "y": -0.995,
          "x": 15
        },
        "6,2",
        {
          "x": 1
        },
        "6,4"
      ],
      [
        {
          "y": -0.88,
          "x": 5.5
        },
        "0,5",
        "0,6",
        {
          "x": 5.5
        },
        "6,0",
        "6,1"
      ],
      [
        {
          "y": -0.875,
          "x": 0.5
        },
        "0,0",
        "0,1"
      ],
      [
        {
          "y": -0.995,
          "x": 18
        },
        "6,5",
        "6,6"
      ],
      [
        {
          "y": -0.38,
          "x": 3.5
        },
        "1,3",
        {
          "x": 11.5
        },
        "7,3"
      ],
      [
        {
          "y": -0.875,
          "x": 2.5
        },
        "1,2",
        {
          "x": 1
        },
        "1,4"
      ],
      [
        {
          "y":-1,
          "x": 15
        },
        "7,2",
        {
          "x": 1
        },
        "7,4"
      ],
      [
        {
          "y": -0.8799999999999999,
          "x": 5.5
        },
        "1,5",
        "1,6",
        {
          "x": 5.5
        },
        "7,0",
        "7,1"
      ],
      [
        {
          "y": -0.875,
          "x": 0.5
        },
        "1,0",
        "1,1"
      ],
      [
        {
          "y":-1,
          "x": 18
        },
        "7,5",
        "7,6"
      ],
      [
        {
          "y": -0.3799999999999999,
          "x": 3.5
        },
        "2,3",
        {
          "x": 11.5
        },
        "8,3"
      ],
      [
        {
          "y": -0.875,
          "x": 2.5
        },
        "2,2",
        {
          "x": 1
        },
        "2,4"
      ],
      [
        {
          "y":-1,
          "x": 15
        },
        "8,2",
        {
          "x": 1
        },
        "8,4"
      ],
      [
        {
          "y": -0.8799999999999999,
          "x": 5.5
        },
        "2,5",
        "2,6",
        {
          "x": 5.5
        },
        "8,0",
        "8,1"
      ],
      [
        {
          "y": -0.875,
          "x": 0.5
        },
        "2,0",
        "2,1"
      ],
      [
        {
          "y":-1,
          "x": 18
        },
        "8,5",
        "8,6"
      ],
      [
        {
          "y": -0.3799999999999999,
          "x": 3.5
        },
        "3,3",
        {
          "x": 11.5
        },
        "9,3"
      ],
      [
        {
          "y": -0.875,
          "x": 2.5
        },
        "3,2",
        {
          "x": 1
        },
        "3,4"
      ],
      [
        {
          "y":-1,
          "x": 15
        },
        "9,2",
        {
          "x": 1
        },
        "9,4"
      ],
      [
        {
          "y": -0.8799999999999999,
          "x": 5.5
        },
        "3,5",
        {
          "x": 0.5,
          "c": "#777777",
          "w": 2
        },
        "5,3",
        {
          "x": 2.5,
          "w": 2
        },
        "11,3",
        {
          "x": 0.5,
          "c": "#cccccc"
        },
        "9,1"
      ],
      [
        {
          "y": -0.875,
          "x": 0.5
        },
        "3,0",
        "3,1"
      ],
      [
        {
          "y":-1,
          "x": 18
        },
        "9,5",
        "9,6"
      ],
      [
        {
          "y": -0.3799999999999999,
          "x": 3.5,
          "c": "#777777"
        },
        "4,3",
        {
          "x": 11.5
        },
        "10,3"
      ],
      [
        {
          "y": -0.875,
          "x": 2.5,
          "c": "#cccccc"
        },
        "4,2",
        {
          "x": 1,
          "c": "#777777"
        },
        "4,4"
      ],
      [
        {
          "y":-1,
          "x": 15
        },
        "10,2",
        {
          "x": 1,
          "c": "#cccccc"
        },
        "10,4"
      ],
      [
        {
          "y": -0.8799999999999999,
          "x": 7,
          "h": 1.5
        },
        "5,0",
        {
          "h": 1.5
        },
        "5,1",
        {
          "h": 1.5
        },
        "5,2",
        {
          "x": 0.5,
          "h": 1.5
        },
        "11,4",
        {
          "h": 1.5
        },
        "11,5",
        {
          "h": 1.5
        },
        "11,6"
      ],
      [
        {
          "y": -0.875,
          "x": 0.5
        },
        "4,0",
        "4,1"
      ],
      [
        {
          "y": -1,
          "x": 18
        },
        "10,5",
        "10,6"
      ]
    ]
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by wilba** [comment]
@MiddleMan5 can you address the suggestions from @drashna please

**Review by MiddleMan5 — COMMENTED** [review]
Updated, sorry for the delay

**Review by drashna — APPROVED** [review]

**Comment by Glutnix** [comment]
is there a hold up with this PR? It'd be swell to get this usable.

**Pull request merged**